scripts/create-package.sh

```diff
@@ -0,0 +1,243 @@
+#!/bin/sh
+
+set -x
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright (C) 2020-2021  Sherman Perry
+
```

Comment:
Suggest using `-set eu`, which will make the script abort if a process errors-out or if an undefined variable is used (instead of barrelling on like nothing happened).

All of the Staging scripts tend to use "safe-mode" (and this is generally how the workflows behave too.. they exit on error):

``` text
scripts/dump-preprocessor-defines.sh:set -euo pipefail
scripts/fetch-workflows.sh:set -euo pipefail
scripts/format-commit.sh:set -eu
scripts/git-prehook-whitespace.sh:set -euo pipefail
scripts/has-commits-since.sh:set -euo pipefail
scripts/verify-ctype-algorithm-include-order.sh:set -euo pipefail
scripts/verify-markdown.sh:set -euo pipefail
```

The  `-o pipefail` is supported by bash, which stops the script if a step in a pipe fails, but this protection isn't available in Posix shell (I believe it will keep barrelling on).

If this is the first time you've used these flags, then there's a bit of a learning curve to adjust your scripting style to be more protective (that was my experience).  But it's well worth it. 

For example, checking the number of passed arguments with `if [ "$#" -lt 3 ]; then` instead of using `$1, $2, or $3` straight away (because those variables will be undefined if the user provided no arguments).

Reply:
Only problem with `-e` is it aborts the script on things like `[ test something ] || do if fail`. Would have to use proper if statements there, which isn't a big deal I suppose.

Reply:
> Only problem with -e is it aborts the script on things like [ test something ] || do if fail. Would have to use proper if statements there, which isn't a big deal I suppose.

Well, looking into it, maybe sometimes...

Reply:
I think setting unbound variables `-u` is going to be a PITA, so probably won't do that.

Reply:
Understood; thanks for exploring if it would be easy or not. (was worth a shot :+1: )